### ISSUE TITLE
Split `Server-Timing` query duration

### ISSUE THREAD

**Issue by steve-chavez** [PostgREST/postgrest#3055]
## Problem

Currently there's no way to know how much time the [pre-request](https://postgrest.org/en/stable/references/transactions.html#pre-request) and [tx-settings](https://postgrest.org/en/stable/references/transactions.html#transaction-scoped-settings) are taking.

`Server-Timing` englobes all of them into a single `query` metric:  

```
$ curl 'localhost:3000/projects' -i

HTTP/1.1 200 OK
Server-Timing: jwt;dur=14.6, render;dur=5.7, plan;dur=89.5, query;dur=523.2
```

And the [execution plan](https://postgrest.org/en/stable/references/admin.html#execution-plan) it's only obtained from the main query.

## Solution

Split the `query` metric into `tx-settings`, `pre-request` (only when it's configured) and `main-query`:

```
$ curl 'localhost:3000/projects' -i

HTTP/1.1 200 OK
Server-Timing: tx-settings;dur=x, pre-request;dur=y, main-query;dur=z
```

Basically involves adding timings separately for these steps:

https://github.com/PostgREST/postgrest/blob/a8387e0ad3860e0c6d956fdae9554745e4453f6d/src/PostgREST/App.hs#L253-L256

**Comment by laurenceisla** [comment]
I'm stuck in trying to use [`timeItT`](https://hackage.haskell.org/package/sydtest-0.15.1.1/docs/Test-Syd-Run.html#v:timeItT) with those queries, since they're not IO and are executed as a block inside the Hasql transaction. Would the solution need to be done in SQL using a EXPLAIN or something similar? Or is there a way to do it full Haskell that I'm missing?

**Comment by steve-chavez** [comment]
Right, I was mislead by the `query` name of the metric. Perhaps we should rename it to `transaction`? Then at least it should be possible to infer the chunk of time the main query takes, like:

```
curl localhost:3000/projects -H "Accept: application/vnd.pgrst.plan;options=analyze" -i

HTTP/1.1 200 OK
Server-Timing: jwt;dur=14.1, render;dur=2.9, plan;dur=90.6, query;dur=524.7

Aggregate  (cost=28.00..28.02 rows=1 width=144) (actual time=0.025..0.025 rows=1 loops=1)
  ->  Seq Scan on projects  (cost=0.00..22.00 rows=1200 width=40) (actual time=0.007..0.008 rows=5 loops=1)
Planning Time: 0.061 ms
Execution Time: 0.049 ms
```
